instance/tests/models/factories/server.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+OpenStackServer model - Factories
+"""
+
+# Imports #####################################################################
+
+import factory
+from factory.django import DjangoModelFactory
+from functools import wraps
+from mock import MagicMock, Mock, patch
+
+from instance.models.server import OpenStackServer
+from instance.tests.base import add_fixture_to_object
+from instance.tests.models.factories.instance import OpenEdXInstanceFactory
+
+
+# Functions ###################################################################
```

Comment:
It would be easier and provide better coverage to implement integration tests instead (or in addition). Most of not all the logic of the OpenStack provisioning depends on the OpenStack cluster and there is little value (some value but not much ;-) in functional / unit tests. Here is an example:

https://github.com/dachary/teuthology/blob/wip-6502-openstack-v3/teuthology/openstack/test/openstack-integration.py#L113